src/libs/dutil/WixToolset.DUtil/butil.cpp

```diff
@@ -61,21 +61,16 @@ static HRESULT OpenBundleKey(
 DAPI_(HRESULT) BundleGetBundleInfo(
     __in_z LPCWSTR wzBundleId,
     __in_z LPCWSTR wzAttribute,
-    __out_ecount_opt(*pcchValueBuf) LPWSTR lpValueBuf,
-    __inout_opt LPDWORD pcchValueBuf
+    __deref_out_z LPWSTR* psczValue
```

Comment:
Downside is that it requires strutil and C. Buffers are language-agnostic(er).

Reply:
This is not exposed through the engine. This is a dutil API which means you're already using VS and C++.

Reply:
Yeah but it complicates the translation layer if you want to use it outside of C. Not a big deal.